RELEASE_NOTES.txt

```diff
@@ -1,6 +1,135 @@
+* Added CLIENT_CONNECTION_ATTRIBUTES system view.
+* Added JMX commands invoker.
+* Added ability to configure metric exporter filter from Spring XML.
+* Added cache dumps.
+* Added client session outbound message queue size limit.
+* Added compact footer usage statistics to the idle_verify command.
+* Added descriptions and values to the properties list command.
+* Added extended debug logging of client connections.
+* Added limit support to IndexQuery.
+* Added logging of snapshot check procedure.
+* Added metrics for operations on secondary indexes.
+* Added node Consistent ID to the index validate command output.
+* Added number of partitions processed by the index worker to the output of the index commands.
+* Added performance statistics for IndexQuery.
+* Added performance statistics info about query properties and processed rows count by query.
+* Added support of SslContextFactory to CommandHandler (instead of GridSslBasicContextFactory).
+* Added support of custom SSL factory to the control.sh utility.
+* Added support of custom location for cache dumps.
+* Added support of skipCopies for DumpReader.
+* Added the cache create command to the control utility.
+* Added the multiple nodes argument to the 'indexes_force_rebuild' command.
+* Added the multiple nodes argument to the 'schedule_indexes_rebuild' command.
+* Added topology snapshot info to services system view.
+* Deprecated legacy authorization approach via Security Context.
+* Deprecated rebalance related methods in the ClientCacheConfiguration.
+* Fixed Java thin client socket#open timeout to ClientConfiguration#timeout property.
+* Fixed JavaNioAccess handling in JDK 14 and later.
+* Fixed MappedFile.map for JDK 14 and newer.
+* Fixed NPE during defragmentation OOM exception handling.
+* Fixed NPE during transaction recovery.
+* Fixed NPE on snapshot create operation for in-memory cluster.
+* Fixed StoredCacheData removing when a node filter is set.
+* Fixed WAL logging to an archived segment after node restart.
+* Fixed an issue when IgniteLock cannot be acquired after release on JDK 17.
+* Fixed an issue when TTL isn't set for entries inserted with "insert from select" query.
+* Fixed an issue when a transaction will never become finished on timeout when deadlock detection is disabled.
+* Fixed an issue where continuous queries could produce excessive memory usage.
+* Fixed assertion error in case of empty SQL query.
+* Fixed assertion error on defragmentation.
+* Fixed broken serialization of communication messages due to incorrect GridCacheQueryRequest class marshalling.
+* Fixed cache objects transformation in Java thin client.
+* Fixed cache scan command page size to meet the limit argument.
+* Fixed connectable nodes list for control.sh.
+* Fixed continuous queries backup ack message sending for expired entries.
+* Fixed control-utility module location for slim and LGPL assemblies.
+* Fixed defragmentation can't be configured to use one thread.
```

Comment:
Suggestion: Fixed the defragmentation can't be configured to use one thread issue.